.github/workflows/docs-preview.yml

```diff
@@ -0,0 +1,172 @@
+name: Build and Preview Manual
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, synchronize, reopened, closed]
+    paths:
+      - ".github/workflows/docs-preview.yml"
+      - "modules/**"
+      - "docs/**"
+
+# Defining permissions here passes it to all workflows.
+# https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token
+permissions:
+  contents: write
+  pull-requests: write
+  issues: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-preview:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@main
+      - uses: DeterminateSystems/magic-nix-cache-action@main
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set default git branch (to reduce log spam)
+        run: git config --global init.defaultBranch main
+
+      - name: Build documentation packages
+        run: nix build .#docs-html --print-build-logs
+
+      - name: Deploy to GitHub Pages preview
+        run: |
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          BRANCH_NAME="gh-pages"
+          PREVIEW_DIR="docs-preview-${PR_NUMBER}"
+
+          # Clone the gh-pages branch and move to the preview subdirectory
+          git clone --single-branch --branch $BRANCH_NAME https://github.com/${{ github.repository }} gh-pages
+          cd gh-pages
+
+          mkdir -p $PREVIEW_DIR
+
+          # Copy the build files to the preview subdirectory
+          cp -rvf ../result/share/doc/nvf/* ./$PREVIEW_DIR
+
+          # Configure git to use the GitHub Actions token for authentication
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "actions@github.com"
+
+          # Set the GitHub token for authentication
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
+
+          # Add and commit the changes
+          git add --all
+          git commit -m "Deploy PR #${PR_NUMBER} preview" || echo "No changes to commit"
+          git push --force origin $BRANCH_NAME
```

Comment:
race condition when multiple PRs are updated

maybe a simple exponential retry is good enough for now

```
for i in 0..10 {
  git pull
  if git push then break; else sleep $randomtime; done
}
```

Reply:
Is this not solved by co-currency opts? Or did that affect only page deployments.

Reply:
me dum

yea that should be good enough, though why is --forced needed here?

Reply:
It did fix something, but I can't recall which. Might've been worth documenting, but a tad too late.